core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -363,17 +361,14 @@ private[spark] class ExternalSorter[K, V, C](
    * Merge-sort a sequence of (K, C) iterators using a given a comparator for the keys.
    */
   private def mergeSort(iterators: Seq[Iterator[Product2[K, C]]], comparator: Comparator[K])
-      : Iterator[Product2[K, C]] =
-  {
+      : Iterator[Product2[K, C]] = {
     val bufferedIters = iterators.filter(_.hasNext).map(_.buffered)
     type Iter = BufferedIterator[Product2[K, C]]
-    val heap = new mutable.PriorityQueue[Iter]()(new Ordering[Iter] {
-      // Use the reverse order because PriorityQueue dequeues the max
-      override def compare(x: Iter, y: Iter): Int = comparator.compare(y.head._1, x.head._1)
-    })
+    val heap = new mutable.PriorityQueue[Iter]()(
```

Comment:
Now I am checking for missing comments by `grep`. It is semi-automatic so I am just sharing you the places I have found, here we lost:

```
 // Use the reverse order because PriorityQueue dequeues the max
```

Reply:
Sure, I can restore that, it's minor